### PR TITLE
Anthropic Endpoint Support

### PR DESCRIPTION
**Pull request by Saghen**
Adds support for Anthropic via their official SDK and the newer [Messages API](https://docs.anthropic.com/claude/reference/messages-streaming). Left a todo regarding the `preprompt` which I passed to Anthropic via their `system` prompt option, rather than passing it in during `buildPrompt`. Let me know which would be preferable
https://github.com/huggingface/chat-ui/pull/923/files#diff-ef33885c5572e45b8c3ba8c5d6b489b258dc0a3fdd7bef12556481f71f5cd45fR53-R54

### CODE REVIEW

**Comment by julien-c** [comment]
cool addition!

**Comment by ptrandev** [comment]
ok go off king

File: src/lib/server/endpoints/anthropic/endpointAnthropic.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+				messages: [{ role: "user", content: prompt }],
+				max_tokens: model.parameters?.max_new_tokens,
+				temperature: model.parameters?.temperature,
+				top_p: model.parameters?.top_p,
+				top_k: model.parameters?.top_k,
+				stop_sequences: model.parameters?.stop,
+				// TODO: is it correct to provide the preprompt here since the client
+				// natively supports it? or in the buildPrompt function?
+				system: preprompt,
+			});
```
Comment (nsarrazin) [inline src/lib/server/endpoints/anthropic/endpointAnthropic.ts:56]:
```suggestion
	return async ({ messages, preprompt }) => {
		let system = preprompt;
		if (messages?.[0]?.from === "system") {
			system = messages[0].content;
		}

		const messagesFormatted = messages
			.filter((message) => message.from !== "system")
			.map((message) => ({
				role: message.from,
				content: message.content,
			})) as unknown as {
			role: "user" | "assistant";
			content: string;
		}[];

		let tokenId = 0;
		return (async function* () {
			const stream = anthropic.messages.stream({
				model: model.id ?? model.name,
				messages: messagesFormatted,
				max_tokens: model.parameters?.max_new_tokens,
				temperature: model.parameters?.temperature,
				top_p: model.parameters?.top_p,
				top_k: model.parameters?.top_k,
				stop_sequences: model.parameters?.stop,
				system,
			});
```

			
			
This should do the trick. `buildPrompt` is not needed in this case, it's used for converting an array of message to a formatted string for APIs that don't have a messages interface, but in the case of anthropic looks like you can just pass an array of message directly. So we format it correctly (they don't support `role: "system"`) and pass the system prompt separately in `system`

File: src/lib/server/endpoints/anthropic/endpointAnthropic.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
+import { z } from "zod";
+import { buildPrompt } from "$lib/buildPrompt";
```
Comment (nsarrazin) [inline src/lib/server/endpoints/anthropic/endpointAnthropic.ts:2]:
```suggestion
```

Don't need this anymore 😁

File: package.json [hunk 3]
```diff
@@ -50,6 +50,7 @@
 	},
 	"type": "module",
 	"dependencies": {
+		"@anthropic-ai/sdk": "^0.17.1",
```
Comment (nsarrazin) [inline package.json:53]: `@anthropic-ai/sdk` should be installed optionally, like the other inference specific packages

`npm i --save-optional @anthropic-ai/sdk`

**Comment by Saghen** [comment]
Thanks! Made the changes and rebased onto main

**Review by nsarrazin — APPROVED** [review]
Works great! Thanks for the contribution :rocket:

**Pull request merged**